src/logic/hooks/useEstimateSafeCreationGas.tsx

```diff
@@ -34,26 +60,12 @@ export const useEstimateSafeCreationGas = ({
   const userAccount = useSelector(userAccountSelector)
 
   useEffect(() => {
-    const estimateGas = async () => {
-      if (!addresses.length || !numOwners || !userAccount) {
-        return
-      }
-
-      const gasEstimation = await estimateGasForDeployingSafe(addresses, numOwners, userAccount, safeCreationSalt)
-      const gasPrice = await calculateGasPrice()
-      const estimatedGasCosts = gasEstimation * parseInt(gasPrice, 10)
-      const gasCost = fromTokenUnit(estimatedGasCosts, nativeCoin.decimals)
-      const gasCostFormatted = formatAmount(gasCost)
-
-      setGasEstimation({
-        gasEstimation,
-        gasCostFormatted,
-        gasLimit: gasEstimation,
-      })
+    if (!addresses.length || !numOwners || !userAccount) {
+      return
     }
 
-    estimateGas()
-  }, [numOwners, userAccount, safeCreationSalt, addresses])
+    estimateGas(userAccount, numOwners, safeCreationSalt, addresses)?.then(setGasEstimation)
+  }, [numOwners, safeCreationSalt, addresses.join()])
```

Comment:
Could the `addresses.join()` be moved into a variable and then that variable be watched?

Reply:
Yep, updated 👍